src/Storage/Repository/DataRepository.cs

```diff
@@ -159,6 +160,45 @@ public async Task<List<DataElement>> ReadAll(Guid instanceGuid)
             return dataElements;
         }
 
+        /// <inheritdoc/>
+        public async Task<Dictionary<string, List<DataElement>>> ReadAllForMultiple(List<string> instanceGuids)
+        {
+            Dictionary<string, List<DataElement>> dataElements = new();
+            if (instanceGuids == null || instanceGuids.Count == 0)
+            {
+                return dataElements;
+            }
+
+            foreach (var guidString in instanceGuids)
+            {
+                dataElements[guidString] = new List<DataElement>();
+            }
+
+            QueryRequestOptions options = new QueryRequestOptions()
+            {
+                MaxBufferedItemCount = 0,
+                MaxConcurrency = -1,
+
+                // Do not set PartitionKey as we are querying across all partitions
+            };
+
+            FeedIterator<DataElement> query = Container
+                .GetItemLinqQueryable<DataElement>(requestOptions: options)
+                .Where(x => instanceGuids.Contains(x.Id))
```

Comment:
Id here would be the id of the dataElement rather than the instanceGuid, won't it? 
Thinking this should be 
`.Where(x => instanceGuids.Contains(x.InstanceGuid ))`